### PR TITLE
FastGRNN Full Rank and FastRNN

### PR DESCRIPTION
**Pull request by adityakusupati**
Implementations for USPS10 dataset.

### CODE REVIEW

**Comment by adityakusupati** [comment]
@SachinG007 and @metastableB FastGRNN Full Rank is tested and good. Something is off for FastRNN model. The implementation is correct but I am not able to match the accuracy of the pytorch model. @SachinG007 there should be a python inference code to test this kind of scenarios and debug.

**Comment by adityakusupati** [comment]
@harsha-simhadri LSTM, FastRNN implementation included. 

All the test drivers should be well tested.

File: c_reference/include/utils.h [hunk 1]
```diff
@@ -124,4 +124,11 @@ void softmax(const float* const input, const unsigned len, float* const ret) {
     ret[i] = expf(input[i] - offset);
 }
 
+// Print a vector of length len
```
Comment (harsha-simhadri) [inline c_reference/include/utils.h:127]: Do not leave any print statements in the header file

File: c_reference/include/utils.h [hunk 2]
```diff
@@ -47,15 +47,15 @@ void v_sigmoid(const float* const vec, const unsigned len, float* const ret) {
 }
 
 void v_tanh(const float* const vec, const unsigned len, float* const ret) {
-  for (unsigned i = 0; i < len; i++) ret[i] = tanhyperbolic(vec[i]);
+  for (unsigned i = 0; i < len; i++) ret[i] = tanh(vec[i]);
```
Comment (harsha-simhadri) [inline c_reference/include/utils.h:50]: we are using tanhyperbolic defined in utils.h file, instead of tanh from math.h, please revert this change

**Comment by harsha-simhadri** [comment]
@adityakusupati  I had to change the RNN signature to allow RNNpool to accept different RNN signatures, please see the latest commit to c_reference branch. Would you mind modifying to reflect the new format. Sorry for this change. Also, would be good to add drivers before merging the PR.
